### PR TITLE
Dependencies update

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Update dependencies to their latest versions (except for `ws` which does not support node.js 6 anymore).

With the new major version of eslint, new rules have been added. Two triggered changes/fixes in the code:

* [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins)
* [require-atomic-update](https://eslint.org/docs/rules/require-atomic-updates) (possible false positive but I "fix" it anyway)

# Other changes

Fixed a minor bug where a non-boolean `multivalued.value` property could be passed to the validation module.

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L822](https://github.com/kuzzleio/kuzzle/blob/66f814b6b9fbe40a8ae93083603637e2f0f85874/lib/api/core/plugins/pluginsManager.js#L822): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L822](https://github.com/kuzzleio/kuzzle/blob/b2da5a463e8d4e25dd5f072f0a3be434c1b4841d/lib/api/core/plugins/pluginsManager.js#L822): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L822](https://github.com/kuzzleio/kuzzle/blob/7a97951b1a75a55502ba63fd15dfe64b977c926d/lib/api/core/plugins/pluginsManager.js#L822): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/services/elasticsearch.js [hunk 1]
```diff
@@ -206,7 +206,7 @@ class ElasticSearch extends Service {
 
     return this.client.get(esRequest)
       .then(result => {
-        if (result._source && 
+        if (result._source &&
```
Comment (Yoann-Abbes) [inline lib/services/elasticsearch.js]: Should not be
```
if (
result._source
&& result._source._kuzzle_info
&& !result._source._kuzzle_info.active
&& !request.input.args.includeTrash
) {
```
?

Comment (scottinet) [reply lib/services/elasticsearch.js]: I didn't change this: the only change is the trailing space that has been removed by my editor.
This code predates the new wrapping rules, so it shouldn't be blocking this PR.

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L819](https://github.com/kuzzleio/kuzzle/blob/50231a6fb512872e43dc56a360b0c33d9c7f5052/lib/api/core/plugins/pluginsManager.js#L819): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**